pkg/nfd-master/nfd-master.go

```diff
@@ -306,6 +344,10 @@ func updateNodeFeatures(helper apihelper.APIHelpers, nodeName string, labels Lab
 		removeLabels(node, oldLabels)
 	}
 
+	// Resolve publishable extended resources
+	resourceActions := getExtendedResourceActions(node, extendedResources)
+	annotations["extended-resources"] = strings.Join(resourceActions.annotations, ",")
```

Comment:
Move the addition of annotations to `SetLabels()` where "label annotations" are added, too. You should be able to use keys directly from `extendedResources`(?)

Reply:
The annotations are added here, because the logic for figuring out which extended resources (annotations) there are going to be after patching, needs to be run. That logic requires the node object. The node instance does not exist prior to this function. Hence the annotations for extended resources are added here, after the necessary node instance is created. Now, do you want me to try to move the creation of the node instance elsewhere and pass it around, or is this ok, as it is?

Reply:
Ah, you mean the filtering out of invalid values. Maybe that should be already earlier, in `filterFeatureLabels()`(?)

Reply:
filterFeatureLabels doesn't sound any better, the needed node instance doesn't exist yet there.

In order to know what the annotations will be, the whole logic currently in getExtendedResourceActions needs to be run. Surely *that* doesn't belong into a function "filterFeatureLabels" which up till my patch has mostly been nicely and cleanly concentrating in doing what the name says - filtering out feature labels. Now up to this point what I've done in filterFeatureLabels is filtering out the extended resources from the labels (plus setting the values from labels to extended resources, but that is a bit of a kludge) which is pretty well still in line with what the function was probably designed for.

I could move the calling of getExtendedResourceActions (which I'll rename probably since changes were requested) into the function "SetLabels" but that would require moving the creation of the node instance into there also, and the resulting patch array and that node instance will need to be moved as arguments of updateNodeFeatures. Do you want that? If you do, I'll do it. Then you will have the annotations set at the same time at the same place where "label annotations" are added. Extended resource logic and patch forming will run before that, and node instance needs to be created before that so that the logic gets a hold of what the extended resources were previously.

Reply:
But, you wouldn't need the node instance for that(?) Everything that's in ´extendedResources` get's added (minus the invalid values) so you could just keys from the map.